### PR TITLE
fix(test): forge init --force does not work correctly

### LINKED ISSUE #803: `forge init --force` does not work correctly
running `forge init --force` in an existing directory fails now if `lib/ds-test` already exists (as it should if you run `forge init && forge init --force`).

Problematic line: https://github.com/gakonst/foundry/blob/0f9fa92e623f079b32e4c4256ac4844020297bf3/cli/src/cmd/init.rs#L124-L125

We should add a check: if we're in a git repo, check if `lib/ds-test` exists, if it doesn't, we install, otherwise we call it with no dependencies (to make sure deps are at least pulled properly)

_Originally posted by @onbjerg in https://github.com/gakonst/foundry/issues/779#issuecomment-1049138241_

### PR DESCRIPTION
**Pull request by SusheendharVijay**
## Motivation

Attempting to fix #803
I don't think it works just yet, wanted to get feedback
 

## Solution

- check if file exists, yes -> pass no deps, else-> install

### CODE REVIEW

File: cli/src/cmd/init.rs [hunk 1]
```diff
@@ -121,8 +121,26 @@ impl Cmd for InitArgs {
...
+                        if Path::new(&root.join("lib/ds-test")).exists() {
+                            println!("\"lib/ds-test\" already exists, skipping install....");
+                            install(&root, vec![], opts)?;
+                        } else {
+                            Dependency::from_str("https://github.com/dapphub/ds-test")
+                                .and_then(|dependency| install(&root, vec![dependency], opts))?;
+                        }
+                    }
+                    false => eyre::bail!("Current working directory is not a git repo!"),
+                }
```
Comment (mattsse) [inline cli/src/cmd/init.rs:143]: use `if is_git.success(){...} else {...}` instead

**Review by mattsse — CHANGES_REQUESTED** [review]
thanks!
smol nit

File: cli/src/cmd/init.rs (hunk 1 above)
Comment (SusheendharVijay) [reply cli/src/cmd/init.rs:143]: will do, any specific reason for this? just curious 😄

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
test case failing `can_init_no_git` apparently this no longer works with no git or is the test case now outdated?

**Comment by SusheendharVijay** [comment]
> test case failing `can_init_no_git` apparently this no longer works with no git or is the test case now outdated?

the tests are correct, I've made a mistake, will patch it soon

**Comment by SusheendharVijay** [comment]
I've removed the `is_git` check, it's not required afaict. tests passing now @mattsse

**Review by mattsse — APPROVED** [review]
thanks, lgtm!

**Pull request merged**